System/Random.hs

```diff
@@ -427,17 +436,17 @@ genByteString n g = runPureGenST g (uniformByteStringPrim n)
 --
 -- @since 1.2
 runPureGenST :: RandomGen g => g -> (forall s . PureGen g -> StateT g (ST s) a) -> (a, g)
-runPureGenST g action = runST $ runGenStateT g $ action PureGen
+runPureGenST g action = runST $ runGenStateT g $ action PureGenI
 {-# INLINE runPureGenST #-}
 
 
 -- | An opaque data type that carries the type of a pure generator
-data PureGen g = PureGen
+data PureGen g = PureGenI
```

Comment:
Just curious - why the constructor name change?

Reply:
Because `PureGen` is a good name for a constructor for frozen version. And I couldn't come up with a better name other than `PureGenI` (Same applies to `MutGenI` and `PrimGenI`)
```haskell
  newtype Frozen (PureGen g) = PureGen g
  thawGen (PureGen g) = PureGenI <$ put g
  freezeGen _ = fmap PureGen get
```

Reply:
On other hand it breaks convention on using same name for type and only constructor. When I see `PureGen` I would expect that it's constructor `PureGen`

Reply:
I am not particularly set on the name. We can refactor it later to something else. It's not gonna be a problem, most important here is the concept.

Reply:
Neither do I. It just looked surprising